### DISCUSSION TITLE
PyTorch Lightning Integration

### DISCUSSION THREAD

**Discussion by vlomonaco** [ContinualAI/avalanche#388] [Ideas]
Someone asked me how _Avalanche_ could work / may be integrated into [Pytorch Lighting](https://github.com/PyTorchLightning/pytorch-lightning). 

As we started reasoning with other important tools such as Sacred #198, we should take Lighting into consideration as well. 
Let's start a discussion about this in this thread. We still need to consolidate the core of avalanche but it may be important to know where we should be going to support this!

**Comment by suiluj** [comment]
Just for cross referencing. Perhaps Pytorch Lightning integration enables support for IterableDatasets which could be useful for some people:

- Discussion about IterableDatasets: https://github.com/ContinualAI/avalanche/discussions/970
- Lightning support IterableDatasets: https://pytorch-lightning.readthedocs.io/en/stable/guides/data.html#accessing-dataloaders-within-lightningmodule
- Multiple dataloaders for multiple tasks in Pytorch Lightning: https://github.com/PyTorchLightning/pytorch-lightning/discussions/11724
- Another Pytorch Lightning discussion about best practice for continual learning https://forums.pytorchlightning.ai/t/whats-the-best-practice-for-continual-learning/250

**Comment by AntonioCarta** [comment]
The problem that we have right now is that none of the maintainers use pytorch lightning. Personally, it's not clear to me if PL would bring more advantages or more problems.

**Reply by suiluj** [reply]
ah okay good to know. I like the callback approach of Avalanche and the special features for continual learning. PL works with similar callbacks and supports IterableDatasets. At the moment I'm not sure which to prefer.

So I will keep an eye on both frameworks. :)
